uiServices/src/services.go

```diff
@@ -96,5 +94,141 @@ func validateJsonWithSchema(config []byte) bool {
         }
     }
     return true
+}
+
+func getConfigs(rw http.ResponseWriter, req *http.Request){
+
+    configPathDir := req.Header.Get("configPathDir")
+    configName := chi.URLParam(req, "configName")
+
+    if !strings.HasSuffix(configPathDir, "/"){
+        configPathDir = configPathDir + "/"
+    }
+
+    file, err := os.Open(fmt.Sprintf("%s%s.xml", configPathDir, configName))
+    if len(configPathDir) <= 1{
+        rw.WriteHeader(http.StatusBadRequest)
+        logrus.Error("No Header Path Found")
+        return
+    }
+    if err != nil {
+        logrus.Error("Configuration Name Not Found: "+configPathDir + configName)
+        rw.WriteHeader(http.StatusNotFound)
+        return
+    }
+    defer file.Close()
+
+    if err != nil {
+        logrus.Error("Configuration Name Not Found: "+configPathDir + configName)
+        rw.WriteHeader(http.StatusNotFound)
+        return
+    }
+
+    var config perfTestUtils.Config
+
+    byteValue, err := ioutil.ReadAll(file)
+    if err != nil{
+        rw.WriteHeader(http.StatusInternalServerError)
+        logrus.Error("Cannot Read File")
+        return
+    }
+
+    err = xml.Unmarshal(byteValue, &config)
+    if err != nil{
+        rw.WriteHeader(http.StatusInternalServerError)
+        logrus.Error("Cannot Unmarshall")
+        return
+    }
+ 
+    configJson, err := json.MarshalIndent(config,"","")
+    if err != nil {
+        rw.WriteHeader(http.StatusInternalServerError)
+        logrus.Error("Cannot Marshall")
+        return
+    }
+
+    rw.WriteHeader(http.StatusOK)
+    rw.Write(configJson)
+    logrus.Println(string(configJson))
 
 }
+
+func putConfigs(rw http.ResponseWriter, req *http.Request) {
+    path := req.Header.Get("configPathDir")
+
+    if !strings.HasSuffix(path, "/") {
+        path = path + "/"
+    }
+
+    if len(path) <= 1 {
+        logrus.Error("No Path Entered")
+        rw.WriteHeader(http.StatusBadRequest)
+        return
+    }
+
+    configName := chi.URLParam(req, "configName")
+
+    if len(configName) <= 1 {
+        logrus.Error("No File Name Entered")
+        rw.WriteHeader(http.StatusNotFound)
+        return
+
+    }
+
+    configPathDir := fmt.Sprintf("%s%s.xml", path, configName)
+    buf := new(bytes.Buffer)
+    buf.ReadFrom(req.Body)
+
+    if !validateJsonWithSchema(buf.Bytes()) {
+        rw.WriteHeader(http.StatusBadRequest)
+        return
+    }
+
+    config := perfTestUtils.Config{}
+    err := json.Unmarshal(buf.Bytes(), &config)
+    if err != nil {
+        rw.WriteHeader(http.StatusBadRequest)
+        return
+    }
+
+    if !FilePathExist(configPathDir) {
+        logrus.Error("File path does not exist", err)
+        rw.WriteHeader(http.StatusConflict)
+        return
+
+    }
+
+    if configName != config.APIName {
+        logrus.Error("Api Name must match File Name")
+        rw.WriteHeader(http.StatusBadRequest)
+        return
+    }
+    //cannot have update with invalid or empty values
+    if config.APIName == "" ||
```

Comment:
is this reachable if we validate against the json schema? I would suggest we re-evaluate this section.